doctor/scutil/scutil_darwin.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package scutil
+
+import (
+	"context"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"tailscale.com/types/logger"
+)
+
+func (Check) Run(ctx context.Context, logf logger.Logf) error {
+	cmd := exec.CommandContext(ctx, "scutil", "--dns")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		logf("error running scutil --dns: %v", err)
+		return nil
+	}
+
+	parsed, err := parseScutilDns(logf, string(out))
+	if err != nil {
+		logf("error parsing scutil --dns output: %v", err)
+		return nil
+	}
+
+	for _, section := range parsed.Sections {
+		logf("section: %s", section.Name)
+		for _, entry := range section.Entries {
+			logf("  entry: %s", entry.Name)
+			for key, val := range entry.Config {
+				logf("    %s=%q", key, val)
+			}
+			for key, list := range entry.ListConfig {
+				for i, val := range list {
+					logf("    %s[%d]=%q", key, i, val)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+type dnsInfo struct {
+	Sections []*dnsSection
+}
+
+type dnsSection struct {
+	Name    string
+	Entries []*dnsEntry
+}
+
+type dnsEntry struct {
+	Name       string
+	Config     map[string]string
+	ListConfig map[string][]string
+}
+
+var (
+	reSpacePrefix = regexp.MustCompile(`\A\s+[^\s]`)
+	reNumSuffix   = regexp.MustCompile(`\A(.+)\[(\d+)\]\z`)
+)
+
+func parseScutilDns(logf logger.Logf, data string) (*dnsInfo, error) {
```

Comment:
Nit: I think it's common Go style for acronyms to remain all caps, and not be treated as words (https://github.com/golang/go/wiki/CodeReviewComments#initialisms), so this should be called `parseScutilDNS`